### PR TITLE
test: added an edge case in ReactionSystem

### PR DESCRIPTION
**Pull request by sivasathyaseeelan**
I have added an edge cases to test ReactionSystem which were not tested before.

## Checklist

- [X] Appropriate tests were added
- [X] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [X] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by TorkelE** [comment]
The test is well written, but I don't see how this tests something relevant to the package? I.e. if I understand it correct, what we want to test is that if the user updates the `Catalyst.forbidden_symbols_error` list with a new forbidden variable, then these cannot be used?

I am mostly surprised that the user can update `Catalyst.forbidden_symbols_error`, which really isn;t something anyone should do.

Is this looking at a specific line in the code which we do not have coverage on? In that case we could have al ook at it to see if there is a better way to cover it.

Even if we do not end up using this, still thanks for your effort with the PR!

File: test/reactionsystem_core/reactionsystem.jl [hunk 1]
```diff
@@ -612,6 +612,31 @@ let
...
 
+# Test for forbidden symbols
+let
+    # Define the forbidden symbols (mocked for the purpose of the test)
+    forbidden_symbols_error = Set([:forbidden1, :forbidden2, :forbidden3])
+    Catalyst.forbidden_symbols_error = forbidden_symbols_error
+
+    # Define parameters and species including forbidden symbols
+    @parameters p1 forbidden1
+    @species S1 forbidden2(t)
```
Comment (TorkelE) [inline test/reactionsystem_core/reactionsystem.jl:623]: Should be `S(t)`, species should always depend on an independent variable.

**Comment by sivasathyaseeelan** [comment]
> The test is well written, but I don't see how this tests something relevant to the package? I.e. if I understand it correct, what we want to test is that if the user updates the `Catalyst.forbidden_symbols_error` list with a new forbidden variable, then these cannot be used?
> 
> I am mostly surprised that the user can update `Catalyst.forbidden_symbols_error`, which really isn;t something anyone should do.
> 
> Is this looking at a specific line in the code which we do not have coverage on? In that case we could have al ook at it to see if there is a better way to cover it.
> 
> Even if we do not end up using this, still thanks for your effort with the PR!

I thought of creating some random forbidden symbols. I will remove it and replace it with any exiting forbidden symbol

Also this covers the below condition 
Screenshot from 2025-01-15 01-54-47

**Comment by TorkelE** [comment]
sounds good

**Comment by TorkelE** [comment]
Also, you might want to use `test_throws` to check whether a line of code generates an error.

**Comment by TorkelE** [comment]
https://docs.julialang.org/en/v1/stdlib/Test/#Test.@test_throws

**Comment by TorkelE** [comment]
Looking closer, this seems like a repeat of
```julia
    # System using a forbidden symbol.
    @test_throws Exception rs = ReactionSystem([Reaction(Γ, [X1], [])], t; named = :rs)
```

from the test block just above yours. 

However, there is an error in that test that means that that specific error is not hit (`named` should be `name`). if you want you can fix that bug error though :)

**Comment by sivasathyaseeelan** [comment]
> Looking closer, this seems like a repeat of
> 
> ```julia
>     # System using a forbidden symbol.
>     @test_throws Exception rs = ReactionSystem([Reaction(Γ, [X1], [])], t; named = :rs)
> ```
> 
> from the test block just above yours.
> 
> However, there is an error in that test that means that that specific error is not hit (`named` should be `name`). if you want you can fix that bug error though :)

I will fix the bug in previous test and remove the new test set

**Comment by TorkelE** [comment]
Thanks, will merge once everything passes.
